### ISSUE TITLE
Alert queries fail with "input data must be wide series but got type not (input refid)"

### ISSUE THREAD

**Issue by opub** [grafana/grafana#46429]
I am trying to create alerts in Grafana 8.4.3 when the number of errors in my CloudWatch logs exceed a certain threshold over a time period. I would normally filter my logs to just errors with `filter @@level="error"`. However almost every query I attempt to run fails with `Failed to evaluate queries and expressions: failed to execute conditions: input data must be a wide series but got type not (input refid)`. I have even tried some of the queries from the "CloudWatch Logs cheat sheet" that appear in the help window.

This one example works.
```
filter @message like /Exception/ |
stats count(*) as exceptionCount by bin(5m) |
sort exceptionCount desc
```

But the others I try to construct in including this one from the help window fail.
```
fields @timestamp, @message |
sort @timestamp desc |
limit 25
```
 
![Screen Shot 2022-03-10 at 3 37 10 PM](https://user-images.githubusercontent.com/3865761/157751000-5fa3d2b7-c97e-4509-9dff-18103af68caa.png)

**What happened**:
Ran example and custom CloudWatch Log queries and they fail with "wide series" error.

**What you expected to happen**:
Query returns all matching log entries and allows me to define alerts based on them.

**How to reproduce it (as minimally and precisely as possible)**:
See sample query in screenshot that was taken directly from the inline help.

**Anything else we need to know?**:
This doesn't appear to be a CloudWatch or AWS permission issue since I am able to query record counts in other CloudWatch Log based dashboards.
![Screen Shot 2022-03-10 at 3 54 43 PM](https://user-images.githubusercontent.com/3865761/157752939-26b8556c-79a2-4882-aa6d-20096978b5f1.png)

**Environment**:
- Grafana version: 8.4.3
- Data source type & version: CloudWatch Logs
- OS Grafana is installed on: Amazon Linux 2
- User OS & Browser: macOS 12.2.1 + Chrome 99.0.4844.51
- Grafana plugins: none

**Comment by Elfo404** [comment]
Hi @opub thanks for reporting this.

Alerting works when the underlying query returns numeric data; from [Grafana's CloudWatch docs](https://grafana.com/docs/grafana/latest/datasources/aws-cloudwatch/#alerting): 
>Since CloudWatch Logs queries can return numeric data, for example through the use of the stats command, alerts are supported. 

in your example, your first query is making use of the `stats` command (therefore returns numeric data) while the second (i'm assuming) is returning raw log entries, on which Grafana cannot do alerting.

Closing this as it seems to be working as expected, please feel free to reopen if you think i missed something

**Comment by opub** [comment]
Thanks @Elfo404! This addresses my issue. I do have a couple recommendations though.

The documentation could definitely be improved in this area. It wasn't clear that only queries that returned numeric data were supported. In my case I wanted the alert to be based on the count of the returned items. Since the conditions include "count()" I expected that to be able to convert any result set to the required numeric value.

Had the error been "query must return numeric values" or similar instead of "input data must be wide series but got type not (input refid)" I probably would have been able to figure this out.

Also the "CloudWatch Logs cheat sheet" that is displayed on the "Create alert rule" page shouldn't include examples that aren't valid for alerts.

**Comment by Elfo404** [comment]
that's indeed true, would you mind creating a separate issue for the docs bit pointing out what is missing/confusing at https://github.com/grafana/grafana/issues/new?title=Docs%20feedback:%20/datasources/aws-cloudwatch/_index.md ?

thanks a lot! 🙏

**Comment by JoshuaKahn** [comment]
Found this issue (I'm getting the same error) and wanted to link from my community post:
https://community.grafana.com/t/no-longer-able-to-create-alert-on-cloudwatch-raw-log-output-in-grafana-8/66928

Is it not possible to create (unified) alerts using Cloudwatch Logs? It was possible in legacy alerting...

**Comment by LevkovychBorya** [comment]
@Elfo404 Hi, even the example doesn't work in my case.
<img width="1440" alt="Screenshot 2022-06-24 at 15 51 54" src="https://user-images.githubusercontent.com/38095999/175550206-a14a3da3-e441-4a65-bae0-3fbdfa8fb2ae.png">

**Comment by Elfo404** [comment]
@JoshuaKahn does this section of the docs help? https://grafana.com/docs/grafana/latest/datasources/aws-cloudwatch/#alerting

**Comment by JoshuaKahn** [comment]
> @JoshuaKahn does this section of the docs help? https://grafana.com/docs/grafana/latest/datasources/aws-cloudwatch/#alerting

Unfortunately not, since I'm referring more to queries that return more than numerical data. Legacy Grafana was able to return these queries and have alerts on them with no issue, but Unified Grafana cannot handle such a query.

Even if I make a query that returns numerical data in the form specified, it's not completely useful as I'd like to see more information associated with that numerical data. For example, a query that returns stats on the number of 'error's within a certain bin size, but also the `loggroup` it came from and the (string) value of the `error` itself will fail because there's more than a number present.

**Comment by ivanstillfront** [comment]
I have the same issue. My query returns valid numeric data that can be printed in a Time series panel but the alerting rule fails with the error: 

```Failed to evaluate queries and expressions: failed to execute conditions: input data must be a wide series but got type long (input refid)```

I am expecting a multidimensional alert just like the video shows. 
My query:

```
fields @timestamp, @message
| parse @message '"* * *" * *' as method, path, version, code, length
| filter code =~ /4\d\d/
| filter path like /\/api/
| stats count() by bin(1h), path
```

**Comment by iwysiu** [comment]
Hi! Since the original problem was solved, I'm going to re-close this issue. We don't presently have any plans to support non-numeric alert data, however, please feel free to create a feature request.

**Comment by sherfyehia** [comment]
## What happened: 
- **when i run this sql query below it run successful with dashborad just three column two (int) and one (character) --- when i run same query to create alert it give me [ input data must be a wide series but got type not (input refid) ] **


`SELECT a.inquery_failed , a.Name ,a.Denomination
from 
(SELECT d.Name ,psr.DenominationID as Denomination ,COUNT(CASE WHEN psr.ProviderServiceRequestStatusID = 3 and psr.RequestTypeID = 1 THEN 1 END) AS inquery_failed
FROM ProviderServiceRequests psr 
inner join Denominations d on d.ID = psr.DenominationID
where psr.CreationDate between DATEADD(minute, -30, GETDATE()) and GETDATE()
group by psr.DenominationID ,d.Name
) as a
order by a.Denomination`

## What you expected to happen: it should run with alert

## How to reproduce it (as minimally and precisely as possible): just accept character data with need timeseries

## Anything else we need to know?:

## Environment:
- Grafana version: 9.4.7
- Data source type & version: sql database
- OS Grafana is installed on: Ubuntu 22.04.1 LTS
- User OS & Browser: root
- Grafana plugins: 
- Others:

**Comment by saketbairoliya2** [comment]
@ivanstillfront - How did you deal with this problem? I'm trying to save a data in the format that is specified here but I'm getting the same error you raised up. https://github.com/grafana/grafana/issues/45295

**Comment by iwysiu** [comment]
Hi @saketbairoliya2 , the "input data must be wide series..." error means that the alerts are being passed non-numeric (usually string) data that they can't parse. You may be able to just alter your query to not return the string data, but if your query is returning multiple time series there was a bug related to cloudwatch logs queries with multiple time series adding a string field that was fixed in [this pr](https://github.com/grafana/grafana/pull/67608) in Grafana 10.

**Comment by saketbairoliya2** [comment]
Hi @iwysiu  - I've a format which is very similar to the format that is mentioned in here: https://github.com/grafana/grafana/issues/45295
<img width="275" alt="Screenshot 2023-07-17 at 1 16 20 PM" src="https://github.com/grafana/grafana/assets/5949441/20e422a3-1aa1-4f19-a801-7fcd46635a2e">

A timestamp column, a label column(Which is string format) and a metric (Numeric). This is as per the mentioned format here: https://grafana.com/docs/grafana/latest/alerting/fundamentals/evaluate-grafana-alerts/#alerting-on-numeric-data-1

How would you suggest we move forward with this?

**Comment by iwysiu** [comment]
Hi @saketbairoliya2 what datasource are you having this issue with? If its mySQL, this thread may be helpful: https://community.grafana.com/t/mysql-columns-as-labels-for-alerting/91255/16

**Comment by Quazmoz** [comment]
> Hi! Since the original problem was solved, I'm going to re-close this issue. We don't presently have any plans to support non-numeric alert data, however, please feel free to create a feature request.

There has been a feature request for many years:
https://github.com/grafana/grafana/issues/7690

**Comment by iwysiu** [comment]
Hi @Quazmoz ! Are you still having issues with CloudWatch Logs and alerting? It should be fixed in [version 10.2.1](https://github.com/grafana/grafana/releases/tag/v10.2.1), but feel free to open a new issue if you run into problems
